### PR TITLE
Add note for Composer timeout in contributing.md.

### LINKED ISSUE #1108: Composer consistently hits the 300 second timeout when running first time setup.
# Bug Report

When running the setup on a fresh install (and not having WordPress Develop cached in Composer), Composer consistently hits the 300s timeout due to the large size of WordPress Develop (and probably in cases where internet speeds may be a bit slower).

## Expected Behavior

Running install for the first time should not timeout.

## Actual Behavior

Composer hits the 300s timeout.

## Steps to Reproduce the Problem

1. Delete your Composer cache.
2. Clone this repo.
3. Run `npm install`.

## Log

```
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 35 installs, 0 updates, 0 removals
  - Installing wordpress/wordpress (5.3.2): Cloning 74214d276c

                                                                                                                                                                                                                                                                               
  [Symfony\Component\Process\Exception\ProcessTimedOutException]                                                                                                                                                                                                               
  The process "git clone --no-checkout 'https://github.com/WordPress/wordpress-develop.git' '/Users/me/Developer/XWP/Stream/vendor/wordpress/wordpress' && cd '/Users/me/Developer/XWP/Stream/vendor/wordpress/wordpress' && git remote add composer 'https://github.com/Word  
  Press/wordpress-develop.git' && git fetch composer && git remote set-url origin 'https://github.com/WordPress/wordpress-develop.git' && git remote set-url composer 'https://github.com/WordPress/wordpress-develop.git'" exceeded the timeout of 300 seconds.               
```

### PR DESCRIPTION
**Pull request by kopepasah**
Closes #1108.

# Checklist

- [x] Project documentation has been updated to reflect the changes in this pull request, if applicable.

### CODE REVIEW

**Comment by kopepasah** [comment]
@kaspars, that could be great too, but I didn't want to enforce this on anyone and it only really happens locally when it is not cached and you are developing. I had not looked at the CI tool portion, but are we currently running tests that require WordPress to be installed?

**Comment by kopepasah** [comment]
@kasparsd after review, I can see that it is needed in CI; however, I am hesitant to adding a timeout to the project and enforcing that on users. Perhaps we can export the setting in CI only?

**Comment by kasparsd** [comment]
@kopepasah I think there is no potential harm of bumping the timeout as long as we're not disabling it completely. Something like 10 minutes sounds reasonable.

File: contributing.md [hunk 1]
```diff
@@ -29,6 +29,14 @@ For setups with local Docker environment you don't need Vagrant and VirtualBox.
 4. Visit [stream.local](https://stream.local) and login using `admin` / `password`.
 5. Activate the Stream plugin.
 
+#### Composer Timeout
+
+On the first setup of a fresh install, you may experience Composer 300s timeout when cloning WordPress Develop from source in order to cache. WordPress Develop repo is large and will consistently timeout when it is not cached, so it is recommended to temporarily remove the timeout for the first time installing:
+
+	export COMPOSER_PROCESS_TIMEOUT=0; npm install
```
Comment (kasparsd) [inline contributing.md:36]: Could we remove the `export` prefix to ensure the timeout environment variable value applies only to this one specific command? That will also ensure users don't have to reset the global composer timeout value.

**Comment by kopepasah** [comment]
@kasparsd I like the idea of adding a process-timeout to composer.json and, since you feel it is safe enough, let's drop this PR and move forward with that timeout.

**Pull request closed without merging**